node/node.go

```diff
@@ -556,6 +562,50 @@ func (n *Node) initDatabase() error {
 	if genesisHash != expectedGenesisHash {
 		return fmt.Errorf("db contains invalid genesis hash. DB Genesis: %s Generated Genesis: %s", genesisHash, expectedGenesisHash)
 	}
+
+	ok, err := n.DB.Has(sessionKey)
+	if err != nil {
+		return err
+	}
+
+	if ok {
+		sessionBytes, err := n.DB.Get(sessionKey)
+		if err != nil {
+			return err
+		}
+		session := session{}
+		if err := json.Unmarshal(sessionBytes, &session); err != nil {
+			return fmt.Errorf("failed to unmarshal session metadata: %w", err)
+		}
+
+		n.Log.Warn(
+			"detected previous ungraceful shutdown",
+			zap.Time("timestamp", session.Timestamp),
+		)
+	}
```

Comment:
typically we just directly call `n.DB.Get` and check for `database.ErrNotFound` rather than calling the DB twice